### PR TITLE
Button Icon Unit Test

### PR DESCRIPTION
**Pull request by JeffersonMat**
Link to associated zer0 card:  
[CARD](https://www.notion.so/rational-nomads/FEATURE-Tests-for-Icon-Button-component-741a04c10e5646bb9cf72c351cfbd65c)
-----

## 1. Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] zer0 card has been moved to the Code Review column
- [X] zer0 card has a link to this PR
- [X] A reviewer has been tagged in the zer0 card

## 2. PR type

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## 3. What is the current behaviour?

Button Icon Component didn't have unit tests

## 4. What is the new behaviour?

Added unit test for button Icon component

### CODE REVIEW

**Comment by padala** [comment]
@ericlowturing  Can you please review this?

**Comment by ericlowturing** [comment]
> @ericlowturing Can you please review this?

This PR is for unit testing, but the test suites are not passed. SO I thought it is not ready to review

File: src/components/Buttons/IconButton/IconButton.test.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	});
+	it('Button to have classname coming through props', () => {
+		const { getByTestId } = renderComponent({ className: 'ClassName-Prop' });
+		const container = getByTestId(TEST_ID.CONTAINER);
+		expect(container).toHaveClass('ClassName-Prop');
+	});
+
+	it('Has alt from props alt', () => {
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		const { getByTestId } = renderComponent({ alt: 'alt-img' });
```
Comment (ericlowturing) [inline src/components/Buttons/IconButton/IconButton.test.tsx:63]: You can just use `renderComponent({ alt: 'alt-img' });` and no need eslint comment in that case. It is just for rendering component.

Comment (ericlowturing) [reply src/components/Buttons/IconButton/IconButton.test.tsx:63]: Or 
const icon = getByTestId(TEST_ID.CONTENT);
expect(icon).toBeInTheDocument();

File: src/components/Buttons/IconButton/IconButton.tsx [hunk 2]
```diff
@@ -13,6 +13,11 @@ type IconButtonProps = {
 	disabled?: boolean;
 };
 
+export const TEST_ID = {
+	CONTAINER: 'icon-button-container',
+	CONTENT: 'icon-button-image',
```
Comment (ericlowturing) [inline src/components/Buttons/IconButton/IconButton.tsx:18]: This test_id is not used in testing suits.

I recommend to use the name: `BUTTON: 'icon-button'`, `ICON: 'icon-button-icon'`. `Container` does not make sense to me.

**Review by ericlowturing — COMMENTED** [review]
LGTM. I left some comments about naming test_ids and minor change requests.

File: src/components/Buttons/IconButton/IconButton.test.tsx (hunk 1 above)
Comment (JeffersonMat) [reply src/components/Buttons/IconButton/IconButton.test.tsx:63]: @ericlowturing. Thank you! Done it!

File: src/components/Buttons/IconButton/IconButton.tsx (hunk 2 above)
Comment (JeffersonMat) [reply src/components/Buttons/IconButton/IconButton.tsx:18]: @ericlowturing Thank you, need to work on my naming skills! Cheers. It's done.

**Review by ericlowturing — APPROVED** [review]

**Pull request merged**